### DISCUSSION TITLE
Zephyr Wifi IoT device - whats a good board to start with?

### DISCUSSION THREAD

**Discussion by torntrousers** [zephyrproject-rtos/zephyr#36349] [Q&A]
I'm looking at using Zephyr in our product and so want to make a prototype IoT device. Just something simple like MQTT publishing over WiFi  connecting to a cloud provider like AWS or Azure. What would be a good MCU board to try that with?

Ferkling around with Google and your user mailing list I see things like [this](https://stackoverflow.com/questions/60369066/zephyr-aws-iot-sample-application) which makes it sound dooable but other than that just users asking for similar but getting no replies, eg [this](https://lists.zephyrproject.org/g/users/message/2610) and [this](https://lists.zephyrproject.org/g/users/message/1425) so wonder whats possible today.

Any tips? Is MQTT, TLS and WiFi working enough for this today and if so are there any boards in particular which have good support and are easy to use to get going with?

**Comment by nashif** [comment]
[disco_l475_iot1](https://docs.zephyrproject.org/latest/boards/arm/disco_l475_iot1/doc/index.html) is one, but there are many, either with wifi built in or using shields.

**Reply by torntrousers** [reply]
I don't have that one, I could go order one but before I do that is there a list somewhere of supported boards with WiFi in case I have one, eg as you mentioned an STM32 I have a B-L4S5I-IOT01A - would that work?  
I have lots of ESP32s but all I see in your user mailing list is questions about it not working, so I suppose thats not a good one to try?

**Reply by nashif** [reply]
re ESP32, @sylvioalves  can provide more information. Wifi should be working now.

**Reply by verjus** [reply]
Here is an example with a board named Pitaya Go of [mqtt client](https://github.com/makerdiary/pitaya-go/tree/master/examples/wifi/mqtt_client_example).

I am trying to port it to zephyr, but I am having partial success only.

**Reply by erwango** [reply]
>  I have a B-L4S5I-IOT01A - would that work?

That's basically the new version of disco_l475_iot1.
You'll get WIFI support on recent Zephyr version (mind there was a bug that got fixed recently)
